patches/src/main/kotlin/app/revanced/patches/spotify/misc/lyrics/CustomLyricsPatch.kt

```diff
@@ -0,0 +1,133 @@
+package app.revanced.patches.spotify.misc.lyrics
+
+import app.revanced.patcher.Fingerprint
+import app.revanced.patcher.extensions.InstructionExtensions.addInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.instructions
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.bytecodePatch
+import app.revanced.patcher.patch.stringOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.util.indexOfFirstInstructionOrThrow
+import app.revanced.util.indexOfFirstInstructionReversedOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.MutableMethodImplementation
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction21t
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction35c
+import com.android.tools.smali.dexlib2.iface.instruction.FiveRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethod
+import com.android.tools.smali.dexlib2.immutable.reference.ImmutableMethodReference
+
+
+@Suppress("unused")
+val customLyricsPatch = bytecodePatch(
+    name = "Custom lyrics provider",
+    description = "Make Spotify use a custom lyrics provider.",
+) {
+    compatibleWith("com.spotify.music")
+
+    val lyricsProviderUrlOption = stringOption(
+        key = "lyricsProviderUrl",
+        title = "Lyrics provider URL",
+        description = "The custom lyrics provider URL.",
```

Comment:
Might drop an example here?
```suggestion
        description = "The URL to a custom lyrics provider such as SpotifyMobileLyricsAPI: https://github.com/Natoune/SpotifyMobileLyricsAPI.",
```